aptos-move/mvhashmap/src/unsync_map.rs

```diff
@@ -94,27 +94,28 @@ impl<
             Some(group_map) => GroupReadResult::Size(group_size_as_sum(
                 group_map
                     .iter()
-                    .flat_map(|(t, v)| v.bytes().map(|bytes| (t, bytes))),
+                    .flat_map(|(t, (v, _))| v.bytes().map(|bytes| (t, bytes))),
             )?),
             None => GroupReadResult::Uninitialized,
         })
     }
 
-    pub fn get_value_from_group(&self, group_key: &K, value_tag: &T) -> GroupReadResult {
+    pub fn read_from_group(&self, group_key: &K, value_tag: &T) -> GroupReadResult {
```

Comment:
We need to provide a layout here, so when we read we can set uninitialized layouts to either None or Some?

Reply:
I have layout as input when doing the id -- value exchange for resource groups. Please let me know if it works.